src/annotations/AnnotationManager.php

```diff
@@ -299,6 +306,22 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if ($this->_traitsSupported && $member_name) {
+                $traitAnnotations = array();
+
+                if (isset($file->traitMethodOverrides[$class_name][$member_name])) {
+                    list($traitName, $originalMemberName) = $file->traitMethodOverrides[$class_name][$member_name];
+                    $traitAnnotations = $this->getAnnotations($traitName, $member_type, $originalMemberName);
+                } else {
+                    foreach ($reflection->getTraitNames() as $traitName)
+                    {
+                        $traitAnnotations += $this->getAnnotations($traitName, $member_type, $member_name);
+                    }
+                }
+
+                $annotations = array_merge($traitAnnotations, $annotations);
```

Comment:
What `!isset($file->data[$key])` represents semantically? No such property/method exists in introspected class? So both of these cases work correctly:

1. property defined in parent class > we're introspecting sub-class with a trait that has a property > trait property wins
2. property defined in a class > we're introspecting that class with a trait that as a property > class property wins
?